File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/other_settings/checkstyle/CheckstyleConfigurationFile.java
package com.github.alexandrecarlton.idea.settings.layout.other_settings.checkstyle;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.nio.file.Path;
import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableCheckstyleConfigurationFile.class)
public interface CheckstyleConfigurationFile {
  @Value.Default
  default boolean active() {
    return false;
  }
  String description();

  String file();
}

File: layout/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/other_settings/checkstyle/CheckstyleConfigurationFile.kt
package com.github.alexandrecarlton.idea.settings.dialog.other_settings.checkstyle

import com.fasterxml.jackson.annotation.JsonProperty

data class CheckstyleConfigurationFile(

    @JsonProperty("Active")
    val active: Boolean = false,

    @JsonProperty("Description")
    val description: String,

    @JsonProperty("File")
    val file: String
)
